### ISSUE TITLE
Object keys highlighting slightly incorrect in recent release

### ISSUE THREAD

**Issue by simov** [babel/babel-sublime#414]
![babel-obj](https://user-images.githubusercontent.com/1694112/124361408-c3218d00-dc37-11eb-890a-0b62b119a2fa.png)

1. Why is the dynamic key highlighted?

Technically this is going to be an object key at runtime, but `a` is a variable in your code, and this is a syntax highlighting about your code and not about the runtime.

2. Why destructured keys are being highlighted?

Again these are keys inside an object, but in fact once destructured `a` and `b` are going to be variables in your code, not keys. That's the whole point of the destructuring.

3. After all of this why the shorthand syntax for keys assignment is not hightlighted?

I know this was never working in babel-sublime, but I'm just pointing out that the above two changes just add on top of the other inconsistencies in this syntax.

**Comment by maullerz** [comment]
+1
Wanna open new issue but noticed that.

Seems parser mark constants in destructuring assignment as strings like it is JSON key.

<img width="568" alt="Screenshot 2021-07-03 at 20 54 50" src="https://user-images.githubusercontent.com/5295025/124362950-03d1d400-dc41-11eb-881e-b990bb3bca5a.png">

```
const { foo: { name, desc }, bar } = getObj()
const { foo: fooKey, bar: barKey } = testObject
const result = { foo, bar }
const strObj = { foo: 'first string', bar: 'second string' }
// Array destructuring works well
const [param1, param2] = myArray
```

Found that old issue - https://github.com/Thom1729/Sublime-JS-Custom/issues/89
But using those scopes didn't help.

**Comment by Thom1729** [comment]
Yep, this is a bug. Babel is built using JS Custom's `string_object_keys` option. That option is applying the `string.unquoted` scope much too broadly and should be rewritten. It shouldn't be applied in any of the examples from the first post. There should be a fix in JS Custom 4.1 and babel-sublime 11.1.

**Comment by Thom1729** [comment]
Try v11.1.0-beta.1.

**Comment by maullerz** [comment]
Seems fixed 👍 
<img width="569" alt="Screenshot 2021-07-06 at 09 30 04" src="https://user-images.githubusercontent.com/5295025/124553011-e5d0c300-de3c-11eb-96f7-e3f0e1319e12.png">

**Comment by Thom1729** [comment]
Fixed in v11.1.0.
